File: typescript/src/Basic_Patterns/Behavioural_Patterns/Visitor/Geometric_Shapes/code/elements/Shape.ts
import {Visitor} from "src/Basic_Patterns/Behavioural_Patterns/Visitor/Geometric_Shapes/code/visitors/Visitor";

/** Element Interface */
interface Shape {
    move(x: number, y: number): void;
    draw(): void;
    accept(visitor: Visitor): void;
}

export { Shape };

File: java/src/Basic_Patterns/Behavioural_Patterns/Double_Dispatch/Problem_Showcase/Elements/Shape.java
package Basic_Patterns.Behavioural_Patterns.Double_Dispatch.Problem_Showcase.Elements;

public class Shape implements Graphic {
    protected String id;

    public Shape(String id) {
        this.id = id;
    }

    @Override
    public void draw() {
        System.out.println("Drawing a shape with id " + id + ".");
    }
}
